pkg/apiserver/queryeditor/service.go

```diff
@@ -141,7 +141,11 @@ func (s *Service) runHandler(c *gin.Context) {
 	defer cancel()
 
 	startTime := time.Now()
-	colNames, rows, err := executeStatements(ctx, utils.GetTiDBConnection(c).DB(), req.Statements)
+	sqlDB, err := utils.GetTiDBConnection(c).DB()
+	if err != nil {
```

Comment:
In what case will this error happen?

Reply:
According to https://gorm.io/docs/generic_interface.html. If the underlying database connection is not a *sql.DB, like in a transaction, it will returns error.